app/controllers/application_controller.rb

```diff
@@ -2,4 +2,24 @@ class ApplicationController < ActionController::Base
   include Clearance::Controller
   protect_from_forgery with: :exception
 
+  def has_permission?(obj)
+    if obj.respond_to?(:user_id)
+      obj.user_id == current_user.id || current_user.admin? || obj.users_with_access.include?(current_user)
+    else
+      current_user.admin? || obj.users_with_access.include?(current_user)
+    end
```

Comment:
I'm confused why this is on both branches of the if statement:  `current_user.admin? || obj.users_with_access.include?(current_user)`